source/common/router/scoped_rds.cc

```diff
@@ -101,70 +107,168 @@ ScopedRdsConfigSubscription::ScopedRdsConfigSubscription(
 }
 
 void ScopedRdsConfigSubscription::onConfigUpdate(
-    const Protobuf::RepeatedPtrField<ProtobufWkt::Any>& resources,
+    const Protobuf::RepeatedPtrField<envoy::api::v2::Resource>& added_resources,
+    const Protobuf::RepeatedPtrField<std::string>& removed_resources,
     const std::string& version_info) {
-  std::vector<envoy::api::v2::ScopedRouteConfiguration> scoped_routes;
-  for (const auto& resource_any : resources) {
-    scoped_routes.emplace_back(MessageUtil::anyConvert<envoy::api::v2::ScopedRouteConfiguration>(
-        resource_any, validation_visitor_));
+  bool any_applied = false;
+  std::vector<std::string> exception_msgs;
+  absl::flat_hash_set<std::string> unique_resource_names;
+  envoy::config::filter::network::http_connection_manager::v2::Rds rds;
+  rds.mutable_config_source()->MergeFrom(rds_config_source_);
+
+  // If new route config sources come after the factory_context_.initManager()'s initialize() been
+  // called, that initManager can't accept new targets. Instead we use a local override which will
+  // start new subscriptions but not wait on them to be ready (to not block on main thread).
+  std::unique_ptr<Init::ManagerImpl> overriding_init_manager;
+  if (factory_context_.initManager().state() == Init::Manager::State::Initialized) {
+    // Pause RDS to not send a burst of RDS requests until we start all the new subscriptions.
+    factory_context_.clusterManager().adsMux().pause(
```

Comment:
When/where do we unpause?

Reply:
when all the new subscriptions have their streams built, around line 187.